OTA.md

```diff
@@ -0,0 +1,90 @@
+# Over-The-Air (OTA) Updates
+
+
+> OTA is experimental and in active development. Breaking changes should be expected often. Check this document frequently for updates
+
+
+## Workflow
+
+In app.viam, add the following to the `services` array; you can alternatively add a `generic` service then edit it to match the following
+
+```json
+    {
+      "name": "OTA",
+      "namespace": "rdk",
+      "type": "generic",
+      "model": "ota_service",
+      "attributes": {
+        "url": "",
+        "version": ""
+		}
+	}
+```
+
+
+In the `url` field, enter the url where a firmware will be downloaded from
+   - if using a local endpoint on the same network, remember to use the proper private ip address
+   - for cloud-hosted resources, embedded auth in the url is easiest, we don't support tokens yet
+
+
+The `version` field is equivalent to a `tag` and can be any arbitrary string of up to 128 characters. After successfully applying the new firmware, this `version` will be stored in NVS. This values is compared to that of the latest machine config from app.viam and will trigger the update process.
+
+
+## Requirements
+
+- An esp32 wrover-e with 8MB of flash memory.
+
+## Build Process
+
+## Primer
+
+```
+# ESP-IDF Partition Table
+# Name,   Type, SubType, Offset,  Size, Flags
+nvs,      data, nvs,     0x9000,  0x4000,
+otadata,  data, ota,     0xd000,  0x2000,
+phy_init, data, phy,     0xf000,  0x1000,
+factory,  app,  factory, 0x10000,  1M,
+ota_0,    app,  ota_0,   0x110000, 1M,
+ota_1,    app,  ota_1,   0x210000, 1M,
+
+```
+
+The terms 'firmware' or 'binary' can be a bit generic.
+In this section, we will refer to two types of binaries that can be built.
+1. a Merged Binary
+2. an App Image
```

Comment:
I added links to the latest esp docs regarding it for clarification. App Image Format is what the format it called. I can see how it might be a bit confusing but wanted to keep things as close to official docs as possible so they can find it if they seach `esp app image` or something